app/code/core/Maho/Blog/controllers/Adminhtml/Blog/PostController.php

```diff
@@ -0,0 +1,153 @@
+    {
+        $this->loadLayout()
+            ->_setActiveMenu('cms/blog_posts');
+        return $this;
+    }
+
+    public function indexAction()
+    {
+        $this->_title($this->__('Blog Posts'));
+        $this->_initAction();
+        $this->renderLayout();
+    }
+
+    public function newAction()
+    {
+        $this->_forward('edit');
+    }
+
+    public function editAction()
+    {
+        $this->_title($this->__('Blog Post'));
+
+        $id = $this->getRequest()->getParam('id');
+        $model = Mage::getModel('blog/post');
+
+        if ($id) {
+            $model->load($id);
+            if (!$model->getId()) {
+                Mage::getSingleton('adminhtml/session')->addError(
+                    Mage::helper('blog')->__('This post no longer exists.'),
+                );
+                $this->_redirect('*/*/');
+                return;
+            }
+        }
+
+        $this->_title($model->getId() ? $model->getTitle() : $this->__('New Post'));
+
+        $data = Mage::getSingleton('adminhtml/session')->getFormData(true);
+        if (!empty($data)) {
+            $model->setData($data);
+        }
+
+        Mage::register('blog_post', $model);
+
+        $this->_initAction()
+            ->_addBreadcrumb(
+                $id ? Mage::helper('blog')->__('Edit Post') : Mage::helper('blog')->__('New Post'),
+                $id ? Mage::helper('blog')->__('Edit Post') : Mage::helper('blog')->__('New Post'),
+            )
+            ->renderLayout();
+    }
+
+    public function saveAction()
+    {
+        if ($data = $this->getRequest()->getPost()) {
+            $model = Mage::getModel('blog/post');
+            $id = $this->getRequest()->getParam('id');
+
+            if ($id) {
+                $model->load($id);
+            }
+
+            try {
+                $model->addData($data)
+                    ->setUpdatedAt(Mage::getSingleton('core/date')->gmtDate())
+                    ->save();
+
+                Mage::getSingleton('adminhtml/session')->addSuccess(
+                    Mage::helper('blog')->__('Post was successfully saved'),
+                );
+                Mage::getSingleton('adminhtml/session')->setFormData(false);
+
+                if ($this->getRequest()->getParam('back')) {
+                    $this->_redirect('*/*/edit', ['id' => $model->getId()]);
+                    return;
+                }
+                $this->_redirect('*/*/');
+                return;
+
+            } catch (Exception $e) {
+                Mage::getSingleton('adminhtml/session')->addError($e->getMessage());
+                Mage::getSingleton('adminhtml/session')->setFormData($data);
+                $this->_redirect('*/*/edit', ['id' => $this->getRequest()->getParam('id')]);
+                return;
+            }
+        }
+        $this->_redirect('*/*/');
+    }
+
+    public function deleteAction()
+    {
+        if ($id = $this->getRequest()->getParam('id')) {
```

Comment:
Double check if you can call this endpoint with GET having just the `/id/123/` part in the URL. There's a force use form key class attribute for this otherwise.

Reply:
yeah they is very much draft code, it will probably change a lot, I just pushed some things I did today, it's really not easy (at least for me) to work with a new entity_type... let's see if I manage to make it work...

Reply:
<img width="341" alt="Screenshot 2025-02-10 alle 22 20 33" src="https://github.com/user-attachments/assets/6476e867-7843-4dbe-b8c4-059c7b40a8b5" />

it gets called with the form key so we should be safe :-)